index.js

```diff
@@ -142,57 +93,43 @@ module.exports = function (opts) {
       //but if not, set a short default (as needed in the tests)
       timeout_inactivity = timeout_inactivity || (opts.timers ? 600e3 : 5e3)
 
-      //set all timeouts to one setting, needed in the tests.
-      if(opts.timeout)
-        timeout_handshake = timeout_inactivity = opts.timeout
-
-      var shsCap = (opts.caps && opts.caps.shs) || opts.appKey || appKey
-      var shs = Shs({
-        keys: opts.keys && toSodiumKeys(opts.keys),
-        seed: opts.seed,
-        appKey: toBuffer(shsCap),
-
-        //****************************************
-        timeout: timeout_handshake,
+      var peers = api.peers = {}
 
-        authenticate: function (pub, cb) {
-          var id = '@'+u.toId(pub)
-          api.auth(id, function (err, auth) {
-            if(err) cb(err)
-            else    cb(null, auth || create.permissions.anonymous)
-          })
-        }
-      })
+      var transports = []
 
-      //figure out the local key. take this from the shs plugin,
-      //because we may have only passed in a seed.
-      var id = '@'+u.toId(shs.publicKey)
+      var transforms = [
+        //function () { return shs }
+      ]
 
-      //use configured port, or a random user port.
-      var port = opts.port || 1024+(~~(Math.random()*(65536-1024)))
-      var host = opts.host || nonPrivate.v4 || nonPrivate.private.v4 || '127.0.0.1'
+      var server, msServer, msClient
 
-      var peers = api.peers = {}
+      function createServer () {
+        if(server) return server
+        var suites = []
+        if(transforms.length < 1)
+          throw new Error('secret-stack needs at least 1 transform protocol')
 
-      var server_protocols = [
-        [Net({port: port, host: host}), shs],
-        [Onion({server: false}), shs]
-      ]
-      var client_protocols = server_protocols
-
-      if (opts["tor-only"])
-        client_protocols = [[Onion({server: false}), shs]]
+        transforms.forEach(function (createTransform, instance) {
+          transports.forEach(function (createTransport) {
+            suites.push([
+              createTransport(instance),
+              createTransform({})
+            ])
+          })
+        })
 
-      var msServer = MultiServer(server_protocols, msLogger)
-      var msClient = MultiServer(client_protocols, msLogger)
+        msClient = msServer = MultiServer(suites)
```

Comment:
How about naming `msClient` and `msServer` just `multiserver` since they are the same?